### PR TITLE
[Summary] Add ‘Clean Logs’ Button

### PR DESCRIPTION
**Pull request by Azuer**
Add 'Clean Logs' Button so than we can clean logs when appium server is running

### CODE REVIEW

File: app/renderer/components/ServerMonitor/ServerMonitor.js [hunk 1]
```diff
@@ -71,9 +71,27 @@ class GetRawLogsButton extends Component {
...
+
+  render () {
+    const {serverStatus, clearLogs} = this.props;
+    if (serverStatus !== STATUS_STOPPED) {
+      return <Button className={styles.serverButton}
+              icon="pause-circle-o" 
+              onClick={clearLogs}>CLean Logs</Button>;
+    } else {
+      return null;
+    }
```
Comment (dpgraham) [inline app/renderer/components/ServerMonitor/ServerMonitor.js]: I don't think the `else { return null; }` is necessary

(EDIT: I stand corrected. The render method breaks if it returns undefined)

File: app/renderer/components/ServerMonitor/ServerMonitor.js [hunk 2]
```diff
@@ -154,6 +172,7 @@ export default class ServerMonitor extends Component {
           <div className={`${styles['button-container']}`}>
             <StartSessionButton {...this.props} />
             <StopButton {...this.props} />
+            <ClearLogsButton {...this.props} />
```
Comment (dpgraham) [inline app/renderer/components/ServerMonitor/ServerMonitor.js]: Do you mind posting a screenshot so we can see what this looks like?

File: app/renderer/components/ServerMonitor/ServerMonitor.js [hunk 3]
```diff
@@ -71,9 +71,27 @@ class GetRawLogsButton extends Component {
...
+  static propTypes = {
+    serverStatus: PropTypes.string.isRequired,
+  }
+
+  render () {
+    const {serverStatus, clearLogs} = this.props;
+    if (serverStatus !== STATUS_STOPPED) {
+      return <Button className={styles.serverButton}
+              icon="pause-circle-o" 
+              onClick={clearLogs}>CLean Logs</Button>;
```
Comment (dpgraham) [inline app/renderer/components/ServerMonitor/ServerMonitor.js]: I think the 'delete' icon would be a better choice

**Comment by dpgraham** [comment]
Thanks! One addition is that in the `renderer/actions/ServerMonitor.js` file, `clearLogs` will need to empty the contents of the raw logs:

Current
```javascript
export function clearLogs () {
  return (dispatch) => {
    dispatch({type: LOGS_CLEARED});
  };
}
```

Next
```javascript
import { fs } from 'appium-support';

/* ... */

export function clearLogs () {
  return (dispatch, getState) => {
    const logfilePath = getState().startServer.logfilePath;
    await fs.writeFile(logfilePath, ''); 
    dispatch({type: LOGS_CLEARED});
  };
}
```

File: app/renderer/components/ServerMonitor/ServerMonitor.js (hunk 2 above)
Comment (Azuer) [inline app/renderer/components/ServerMonitor/ServerMonitor.js]: image

Comment (Azuer) [inline app/renderer/components/ServerMonitor/ServerMonitor.js]: image

**Comment by dpgraham** [comment]
I completely forgot that you can't use `fs` on the renderer thread (because it's a browser context). @Azuer you can take that revision out and we'll merge your changes, and I'll just make another PR on top of it that does the raw logs deletion.

**Comment by Azuer** [comment]
Ok, I got it.
I’ve already token it out and rebased changes.

From: Dan Graham
Reply-To: appium/appium-desktop
Date: Thursday, 17 August 2017 at 23:52
To: appium/appium-desktop
Cc: Azuer, Mention
Subject: Re: [appium/appium-desktop] [Summary] Add ‘Clean Logs’ Button (#272)


I completely forgot that you can't use fs on the renderer thread (because it's a browser context). @Azuer<https://github.com/azuer> you can take that revision out and we'll merge your changes, and I'll just make another PR on top of it that does the raw logs deletion.

File: app/renderer/components/ServerMonitor/ServerMonitor.js [hunk 4]
```diff
@@ -71,9 +71,27 @@ class GetRawLogsButton extends Component {
...
 }
 
+class DeleteButton extends Component {
+  static propTypes = {
+    serverStatus: PropTypes.string.isRequired,
+  }
+
+  render () {
+    const {serverStatus, clearLogs} = this.props;
+    if (serverStatus !== STATUS_STOPPED) {
```
Comment (mykola-mokhnach) [inline app/renderer/components/ServerMonitor/ServerMonitor.js]:
```js
return serverStatus === STATUS_STOPPED ? null : <Button className={styles.serverButton}
                                                                                         icon="pause-circle-o" onClick={clearLogs}>Delete</Button>;
```

Comment (Azuer) [inline app/renderer/components/ServerMonitor/ServerMonitor.js]: Ok,I've changed as you suggested and rebased the commit

**Pull request closed without merging**